matrix/validate_sudoku_board.py

```diff
@@ -0,0 +1,101 @@
+"""
+LeetCode 36. Valid Sudoku
+https://leetcode.com/problems/valid-sudoku/
+https://en.wikipedia.org/wiki/Sudoku
+
+Determine if a 9 x 9 Sudoku board is valid. Only the filled cells need to be
+validated according to the following rules:
+
+- Each row must contain the digits 1-9 without repetition.
+- Each column must contain the digits 1-9 without repetition.
+- Each of the nine 3 x 3 sub-boxes of the grid must contain the digits 1-9
+  without repetition.
+
+Note:
+
+A Sudoku board (partially filled) could be valid but is not necessarily
+solvable.
+
+Only the filled cells need to be validated according to the mentioned rules.
+"""
+
+from collections import defaultdict
+
+NUM_ROWS = 9
+NUM_COLS = 9
+EMPTY_CELL = "."
+IS_VALID = True
```

Comment:
Let's use NUM_SQUARES instead of NUM_ROWS and NUM_ROWS.
Lets True and False in a boolean function.
```suggestion
NUM_SQUARES = 9
EMPTY_CELL = "."
```